### PR TITLE
Add basic Site Logos support in anticipation of WP 4.5 release

### PR DESCRIPTION
**Pull request by sixhours**
See #922

### CODE REVIEW

**Comment by jrfnl** [comment]
Would the minimum supported WP version have to go up for this ?

**Comment by sixhours** [comment]
I don't think so. It falls back to nothing if the function doesn't exist; we'd want to remove this backwards compatibility when 4.7 is released, so I added a todo item for that.

**Comment by limestreet** [comment]
Hi,
finally function name is `the_custom_logo()`, support should be added via `add_theme_support('custom-logo')` and we can check conditionally if is set up with `has_custom_logo()`

File: inc/template-tags.php [hunk 1]
```diff
@@ -120,7 +120,6 @@ function _s_category_transient_flusher() {
 add_action( 'edit_category', '_s_category_transient_flusher' );
 add_action( 'save_post',     '_s_category_transient_flusher' );
 
-
 if ( ! function_exists( '_s_the_custom_logo' ) ) :
```
Comment (bahiirwa) [inline inc/template-tags.php]: if ( function_exists( '_s_the_custom_logo' ) ) {
		the_custom_logo();
	}

Comment (sixhours) [inline inc/template-tags.php]: We want it to be `if ( ! function_exists( '_s_the_custom_logo' ) )` to allow child themes to redeclare the function, unless I'm misunderstanding your comment.

**Comment by philiparthurmoore** [comment]
What's keeping this from being merged? I'm happy to code review it if that's what's needed.

**Comment by sixhours** [comment]
I think all it needs is a test and code review. :)

File: functions.php [hunk 2]
```diff
@@ -36,6 +36,12 @@ function _s_setup() {
 	add_theme_support( 'title-tag' );
 
 	/*
+	 * Enable support for site logo.
+	 */
+	add_image_size( '_s-logo', 500, 500 );
+	add_theme_support( 'site-logo', array( 'size' => '_s-logo' ) );
```
Comment (grappler) [inline functions.php]: I would remove the support for Jetpacks logo.

File: functions.php [hunk 3]
```diff
@@ -60,6 +66,16 @@ function _s_setup() {
 	) );
 
 	/*
+	 * Enable support for custom logo.
+	 */
```
Comment (grappler) [inline functions.php]: It would be good to add a link to https://codex.wordpress.org/Theme_Logo like we have done with other features.

**Comment by philiparthurmoore** [comment]
@grappler, thank you.

@sixhours, thank you. :smile:

**Comment by davidakennedy** [comment]
I'm closing this out in favor of a refreshed pull request, #1136. Let's continue the work there.

**Pull request closed without merging**